src/scope-lang/nested-scope-visitor/actions-visitor.ts

```diff
@@ -9,6 +9,7 @@ import * as parser from "../parser";
 const ScopeParser: any = parser.ScopeParser;
 
 // A new parser instance with CST output (enabled by default).
+// CR: why are we creating a new parser instance? are not we already exposing a singleton Parser?
```

Comment:
Sounds legit. I just used the "code at hand", optimisation to follow after I got a prototype working. Thanks